dbt/adapters/redshift/relation_configs/materialized_view.py

```diff
@@ -123,9 +123,25 @@ def parse_model_node(cls, model_node: ModelNode) -> dict:
             "schema_name": model_node.schema,
             "database_name": model_node.database,
             "backup": model_node.config.extra.get("backup"),
-            "autorefresh": model_node.config.extra.get("auto_refresh"),
         }
 
+        autorefresh_value = model_node.config.extra.get("auto_refresh")
+        if autorefresh_value is not None:
+            if isinstance(autorefresh_value, bool):
+                config_dict["autorefresh"] = autorefresh_value
+            elif isinstance(autorefresh_value, str):
+                lower_autorefresh = autorefresh_value.lower()
+                if lower_autorefresh == "true":
```

Comment:
It _feels_ like this should be a utility function shared across core, especially as it looks like in other places we're relying on the soon to be deprecated `distutils`: https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/task/test.py#L54

Reply:
That's fair. In fact, this PR is necessary specifically because we made the decision that we should make it easier on the user by always accepting string versions of boolean configs across the board, not just for `auto_refresh`. Since this needs to be backported to 1.6, are you comfortable with an adapter-specific solution here with a general solution for 1.7 and forward?

Reply:
makes sense to me